__tests__/endpoints.test.js

```diff
@@ -136,6 +136,82 @@ describe("/api/articles/:article_id/comments", () => {
         expect(body.msg).toBe("Bad request");
       });
   });
+  test("POST:201 inserts a new comment to the correct article and sends back the new comment", () => {
+    const newComment = {
+      body: "lol nice",
+      username: "icellusedkars",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  test("POST:400 responds with an appropriate error message when provided with an invalid username (no username)", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        body: "lol nice",
+      })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad request");
+      });
+  });
+  test("POST:400 responds with an appropriate error message when provided with an invalid username", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        body: "lol nice",
+        username: "Mr Nobody",
+      })
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Article/User not found!");
```

Comment:
Best not to have dual error messages, as this can be confusing for debugging. Better to use a more generic message